src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -160,6 +162,33 @@ public sealed class ResolvePackageAssets : TaskBase
         /// </summary>
         public bool DesignTimeBuild { get; set; }
 
+        /// <summary>
+        /// Eg: "Microsoft.NETCore.App;NETStandard.Library"
+        /// </summary>
+        [Required]
+        public string DefaultImplicitPackages { get; set; }
+
+        /// <summary>
+        /// Items with metadata "ParentTarget" and "ParentPackage", which allows determining the hierarchy of package references.
+        /// </summary>
+        [Required]
+        public ITaskItem[] InputPackageDependencies { get; set; }
+
+        /// <summary>
+        /// Information about each package in the project, with metadata:
+        /// - Name = "MetadataExtractor"
+        /// - Path = "metadataextractor/1.0.0"
+        /// - ResolvedPath = "C:\Users\drnoakes\.nuget\packages\metadataextractor\1.0.0"
+        /// - Type = "package"
+        /// - Version = "2.3.0"
+        /// - DiagnosticLevel = ""
+        /// </summary>
+        [Required]
+        public ITaskItem[] PackageDefinitions { get; set; }
```

Comment:
`InputPackageDependencies` and `PackageDefinitions` shouldn't be inputs to this task, they should be read from the .assets file as part of this task.

Since they're required inputs and not currently being passed in, I would expect all builds with this change to fail.

Reply:
The code changed to read InputPackageDependencies and PackageDefinitions from assets file.